### PR TITLE
Display subscribers with correct status [MAILPOET-1944]

### PR DESCRIPTION
**Pull request by pavel-mailpoet**

### CODE REVIEW

File: lib/API/JSON/v1/Subscribers.php [hunk 1]
```diff
@@ -95,24 +95,42 @@ function listing($data = array()) {
...
+
+    return $this->successResponse($result, array(
       'count' => $listing_data['count'],
       'filters' => $listing_data['filters'],
       'groups' => $listing_data['groups']
     ));
   }
 
+  private function fixSubscriberStatus(array $subscriber, $segment_id) {
+    $segment_status = Subscriber::STATUS_SUBSCRIBED;
```
Comment (NeosinneR) [inline lib/API/JSON/v1/Subscribers.php]: Why is this set to `subscribed`? It seems unnecessary.

Comment (pavel-mailpoet) [reply lib/API/JSON/v1/Subscribers.php]: The line is not necessary, but it is here to make sure the variable is always defined and always the same type. But maybe it is confusing and I'll refactor the function a bit.

**Comment by pavel-mailpoet** [comment]
> In the spec, there is mentioned acceptance test, but there is integration. Is that something you agreed on?

Yes, we spoke about it with Michelle and we agreed this is an edge case that can live with an integration test.

**Comment by pavel-mailpoet** [comment]
@michelleshull I am trying to reproduce the problem with WordPress user you described. But I cannot reproduce it. Deleting WordPress users is in the code area that my branch didn't even touch. So it seems unrelated. I rebased master so if there was something in master this could fix it.

**Comment by michelleshull** [comment]
Send it back to me, I'll test again on some different sites.

**Comment by michelleshull** [comment]
I can't reproduce now, either. I think it was related to that user being marked as spam by Akismet, but it's not related to this branch.
